README.md
# phpRechnung

phpRechnung ist ein einfach-zu-benutzendes Rechnungsprogramm.

Mit phpRechnung können Sie folgende Aufgaben erledigen:
Adressbuch und Positionen / Artikel verwalten, Angebote und Rechnungen erstellen
Zahlungen vornehmen, Kassenbuch und offene Posten verwalten ...
Berichte erstellen: Buchungsdetails, Rechnungsausgangsbuch ...

Sprachen:
Deutsch, Englisch, Französisch, Italienisch, Kroatisch, Niederländisch, Polnisch und Spanisch.


Aus Basis  Edy Corak  "Edy's phpRechnung" unter https://www.loenshotel.de/phpRechnung

Letzter Version von "Edy's phpRechnung" war 1.6.5 (Stand 2015-10-09)

Lizenz: GPLv2


## Requirements


### Server:

* Webserver: Apache, Lighttpd, nginX oder auch ( Microsoft IIS - nicht getestet )

* PHP:
	PHP Version 5 - ( mbstring, PEAR::Mail, Mail_Mime )

* DB: 
	MySQL Version 5 oder 4 (oder MariaDB, zukünftig über PDO auch Postgres)

* MailServer:
	Postfix, Exim, Sendmail...

### Client:

* Browser: z. B. Firefox, Konqueror, Safari, Opera, Google Chrome ... ( Javascript muss eingeschaltet sein )

* PDF-Betrachter

## Installation

### Wichtig:
Unter Einstellung muss eine korrekte E-Mail Adresse eingetragen
werden, da ansonsten der E-Mail Versand nicht funktioniert.
Das selbe gilt auch, wenn ein Relay-Server benutzt wird,
da ansonsten die E-Mail's vom Relay-Server abgewiesen werden.

Wenn Sie keinen eigenen E-Mail Server haben, dann stellen Sie
bitte in der Datei /include/phprechnung.inc.php die Variable
$PHPSendMail = ""; auf $PHPSendMail = "1"; ein.

In der Datei /include/mail.inc.php muessen noch Aenderungen vorgenommen werden Host, Port und falls Anmeldung notwendig ist, $Smtp["auth"] auf true zu stellen und $Smtp["username"] und $Smtp["password"] eintragen.

	$Smtp["host"] = "ihrmailserver";
	$Smtp["port"] = "25";
	$Smtp["auth"] = false;
	$Smtp["username"] = "";
	$Smtp["password"] = "";

### Dateien entpacken, kopieren und Berechtigungen setzen

phpRechnung_1_6_4.tar.gz entpacken 
  tar -zxvf phpRechnung_1_6_4.tar.gz

Das Verzeichnis 'phpRechnung_1_6_4' auf den Webserver kopieren, z. B. /var/www Sie koennen das Verzeichnis auch umbenennen.

Seit phpRechnung 1.6 RC1 wird Smarty Template Engine verwendet. Damit das ganze funktioniert, muessen noch einige Aenderungen vorgenommen werden.
	
Der Webserver muss fuer die Verzeichnise '/include/smarty/cache' und '/include/smarty/templates_c' Schreibrechte bekommen.

*Es wird empfohlen ein Verzeichnis außerhalb des DocumentRoot zu verwenden*

Das gilt auch fuer die Verzeichnise '/include/smarty/templates' und '/include/smarty/configs'

Nach den Änderungen muessen Sie dies auch phpRechnung mitteilen.

In '/include/smarty.inc.php' folgende Variablen anpassen:
	$Templateroot und $Cacheroot von $_SERVER['DOCUMENT_ROOT']."phpRechnung";
	
aendern in z. B.:

	$smarty->template_dir = "/var/www/include/smarty/templates";
	$smarty->config_dir = "/var/www/include/smarty/configs";
	$smarty->compile_dir = "/var/www/include/smarty/templates_c";
	$smarty->cache_dir = "/var/www/include/smarty/cache";

Achten Sie bitte auf eine korrekte Schreibweise!


### Datenbank erstellen (Erstinstallation):

Dieser Schritt ist nur fuer eine Neu-Installation gedacht.
Wenn Sie ein UPDATE von phpRechnung durchfuehren möchten, dann lesen Sie bitte unter UPDATE weiter.

	mysqladmin -u 'benutzername' -p create phprechnung

Tabellen erstellen:

	mysql -u 'benutzername' -p phprechnung < phprechnung_1_6_4.sql

Falls kein 'root' Zugriff vorhanden, dann am besten 'phpMyAdmin' benutzen.

Zuerst Datenbank erstellen, waehlen Sie dann die Datenbank aus, dann Importieren,	Datei dursuchen, passende SQL Datei auswaehlen: phprechnung_1_6_4.sql, Zeichencodierung der Datei 'latin1' waehlen und auf OK klicken.

Sie sollten dann z. B. folgende Meldung sehen:
	Der Import wurde erfolgreich abgeschlossen, 127 Abfragen wurden ausgefuehrt.

Folgende Dateien muessen geändert werden, damit man auf die neu erstellte Datenbank 'phprechnung' zugreifen kann:

Für die Datenbankverbindung 
  'include/dbconf.php'
  
	_DBHOST z. B. "localhost"
	_BBUSER z. B. "benutzername"
	_DBPASS z. B. "benutzerpasswort"
	_DBNAME z. B. "phprechnung"

Der _DBUSER muss ueber ausreichende Rechte auf dem Datenbankserver verfuegen - Erstellen, Aendern, Loeschen etc.

	'phprechnung.inc.php'
	
( wichtig ist die Variable "$web" dort bitte eigene Webadresse zu phpRechnung eintragen ) z. B. http://ihrserver/phpRechnung ( ohne '/' am Ende ) da sonst die Weiterleitung nicht funktioniert.

Fuer die Benutzer die phpRechnung auf einem Windows Server betreiben,	bitte unter Konfiguration/Einstellung den Wert von TMP-Verzeichnis anpassen. Standard ist '/tmp/' in z. B. : 
  'c:\windows\temp\'

Bitte beachten Sie den letzten Backslash. Wichtig. Der Webserver muss Schreibrechte fuer das Verzeichnis haben.
	
Auf einem Windows Server bitte auch 'session.save_path' ueberpruefen.

Standard ist session.save_path = /tmp
Wenn das der Fall ist, dann bitte ändern in z. B. 
  session.save_path = c:\windows\temp

### Login:

Administrator: admin, admin

Die Benutzerdaten sind in der Tabelle 'user' verschlüsselt gespeichert

Nach der Ersten Anmeldung werden Sie aufgefordert die phpRechnung Lizenz ( GPL v2 ) zu akzeptieren.
Dieses Vorgehen ist fuer jeden Benutzer einmalig notwendig.

Das Administrator-Passwort kann geändert werden, nur der Anmeldename 'admin' darf nicht verändert werden. Ansonsten werden die Berechtigungen für den Administrator nicht mehr gesetzt!


Bitte haben Sie Verständnis, dass der ursprüngliche Autor Edy Corak < edy at loenshotel dot de > direkte Anfragen zu phpRechnung aus Zeitgründen leider nicht mehr beantworten kann.

Ursprüngliche phpRechnung Homepage - http://www.loenshotel.de/phpRechnung/

@Edy Corak: Vielen Dank für die Arbeit und diese sehr praktische Anwendung!

Vielen Dank fuer die Nutzung von phpRechnung


ChangeLog
/*
	ChangeLog - 06.10.2015

	phpRechnung - is easy-to-use Web-based multilingual accounting software.
	Copyright (C) 2001 - 2015 Edy Corak < edy at loenshotel dot de >

	This program is free software; you can redistribute it and/or modify
	it under the terms of the GNU General Public License as published by
	the Free Software Foundation; either version 2 of the License, or
	(at your option) any later version.

	This program is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	GNU General Public License for more details.

	You should have received a copy of the GNU General Public License
	along with this program; if not, write to the Free Software
	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

	06.10.2015

	Changes in phpRechnung 1.6.5
	---------------------------------------

	This is a Security Bugfix-Only-Release.

	A security issue affects the following phpRechnung releases:

	phpRechnung <= 1.6.4

	It is recommendend to update your phpRechnung <= 1.6.4 to version 1.6.5 as soon as possible.

	JVN#02671769 Link: https://jvn.jp/en/

	list.php of phpRechnung contains a SQL injection (CWE-89) vulnerability.
	An authenticated attacker may obtain or alter information stored in the database.

	31.01.2011

	Changes in phpRechnung 1.6.4
	---------------------------------------

	- FormatDBNumber has been fixed.

	27.01.2011

	Changes in phpRechnung 1.6.3
	---------------------------------------

	- Problem creating initial cashbook entry was fixed.

	- Simple credit note function was added. You can create invoices with negative total amount.
	  By choosing 'bar/cash' as method of payment, cash book will be also automatically updated.

	- If PrintCompanyData was disabled the Company Data was also not included by sending invoice as PDF-Attachment. fixed.

	- minor bugs were fixed

	05.01.2011

	Changes in phpRechnung 1.6.2
	---------------------------------------

	This is a Bugfix-Only-Release.

	- Last Site Redirection missing query was fixed

	- syslog_syslogid_seq was disabled

	- minor bugs were fixed


	08.12.2010

	Changes in phpRechnung 1.6.1
	---------------------------------------

	This is a Security Bugfix-Only-Release.

	A security issue affects the following phpRechnung releases:

	phpRechnung <= 1.6

	There are security vulnerabilities in phpRechnung 1.6 which
	allow an unauthorized user to have read access of the data.

	It is strongly recommendend to update your phpRechnung <= 1.6 to version 1.6.1 as soon as possible.


	03.12.2010

	Changes in phpRechnung 1.6
	---------------------------------------

	This is a Security Bugfix-Only-Release.

	A security issue affects the following phpRechnung releases:

	phpRechnung <= 1.6 RC2

	It is strongly recommendend to update your phpRechnung <= 1.6 RC2 to version 1.6 as soon as possible.

	There are multiple security vulnerabilities in phpRechnung 1.6 RC2 which
	allow an unauthorized user to take control of the software.

	Thank's to Brendan Coles for testing and reporting this security issues.


	- Several warnings have been fixed

	- Improved E-Mail function ( mbstring, PEAR::Mail, Mail_Mime is required )

	- Minor bugs have been fixed

	- FPDF was updated to version 1.6


	12.07.2008

	Changes in phpRechnung 1.6 RC2
	---------------------------------------

	There are no new features in this release.

	This is a Bugfix-Only-Release.

	- Addressbook
	  Fix Detail Search non existing fields removed
	  Fix customer info error if message was missing

	- Reports
	  Fix out of memory

	  Note:
	  When creating very large PDF Documents there is a memory limit

	  I have created reports with 457 pages approx. 17130 entries without any problems
	  php.ini -> memory_limit = 16M;

	- Offer / Invoice
	  Show only active positions

	- Update
	  Rename the directory to /updatetable
	  to prevent errors when unpacking the archive on Windows

	- phprechnung.inc.php
	  Easy change of Number Format ( function Format_Number )

	#2001081 - phpRechnung 1.6 RC1 - Update - Offer Status
	  Fix Offer when updating 'OfferStatus'

	#2001200 - phpRechnung 1.6 RC1 - Update - Addressbook - PrintName
	  Fix Addressbook when updating 'PositionName'

	#2004968 -  phpRechnung 1.6 RC1 - Useradministration - admin
	  Fix Usergroup1, Usergroup2 'admin'

	#2010827 - Position - Group
	  Fix wrong table name

	#2012232 - Addressbook Info - Print
	  Fix old / non existing fields

	16.06.2008

	Changes in phpRechnung 1.6 RC1
	---------------------------------------

	- phpRechnung 1.6 RC1 is a total rewrite.

	- phpRechnung is Valid XHTML 1.0 Transitional

	- phpRechnung now has Dutch translation
		Thank's to Jerry van Kranenburg

	- Smarty Template Engine was included ( 2.6.19 )
		Smarty is released under GNU LGPL
		Copyright 2001-2005 New Digital Group, Inc.

	- ADOdb library was included ( adodb498 )
		ADOdb is released under both BSD license and
		Lesser GPL library license.
		(c) 2000-2005 John Lim (jlim#natsoft.com.my).
		All rights reserved.

	- Many improvements

	- Accesskey's
	-------------

		Alt+W - Startpage
		Alt+A - Addressbook
		Alt+P - Position
		Alt+O - Offer
		Alt+I - Invoice
		Alt+N - Creditnote
		Alt+M - Payment
		Alt+C - Cashbook
		Alt+R - Reports
		Alt+S - Configuration
		Alt+U - Super User
		Alt+L - Logout

		Submenu can be reached by pressing Alt+1, for first entry and so on....
		Will not work if you have more than one tab open.

	- phpRechnung Project Home page http://sourceforge.net/projects/phprechnung/

	09.04.2006

	Changes in phpInvoice 1.6 Test3.01
	----------------------------------

	There are no new features in this release; only small corrections.
	$pdfdir is now set to "/tmp/" in phprechnung.inc.php to prevent errors
	when sending invoices or offers by email.
	Format cleaning was done in some files.

	02.10.2005

	Changes in phpInvoice 1.6 Test3

index.php
<?php

/*
	index.php

	phpRechnung - is easy-to-use Web-based multilingual accounting software.
	Copyright (C) 2001 - 2015 Edy Corak < edy at loenshotel dot de >

	This program is free software; you can redistribute it and/or modify
	it under the terms of the GNU General Public License as published by
	the Free Software Foundation; either version 2 of the License, or
	(at your option) any later version.

	This program is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	GNU General Public License for more details.

	You should have received a copy of the GNU General Public License
	along with this program; if not, write to the Free Software
	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

require_once("include/phprechnung.inc.php");
require_once("include/smarty.inc.php");

CheckUser();
CheckSession();
UserSite();

// Get current date
//
$Today = date('d.m.Y');

// Database connection
//
DBConnect();

// Select all invoices with open accounts
$query = $db->Execute("SELECT PAID, DATE_FORMAT(INVOICE_DATE,'%d.%m.%Y') AS INVOICE_DDATE, CREATEDBY FROM invoice WHERE PAID='2' AND CREATEDBY='$_SESSION[Username]' ORDER BY INVOICE_DATE ASC LIMIT 1");

if (!$query)
{
	$_SESSION['UserReminder'] = "0";
}
else
{
	foreach($query as $f)
	{
		$Date_From = $f['INVOICE_DDATE'];
		$smarty->assign("From_Date","$Date_From");

		$Paid = $f['PAID'];
		$CreatedBy = $f['CREATEDBY'];

		// Explode date and calculate how many days the account is open
		//
		if(!empty($Today))
			list($day1,$month1,$year1)=explode(".",$Today);
		if(!empty($Date_From))
			list($day2,$month2,$year2)=explode(".",$Date_From);

		$date1 = mktime(0,0,0,$month1,$day1,$year1);
		$date2 = mktime(0,0,0,$month2,$day2,$year2);
		$difference = round(($date1-$date2)/86400); // Date today - from / 24h
		$smarty->assign("Difference","$difference");
		$till_date  = date('d.m.Y',mktime(0, 0, 0, date('m'), date('d')-$ReminderDays, date('Y')));
		$smarty->assign("Till_Date","$till_date");

		// Who is logged in ?
		//
		if(isset($_SESSION['Username']) && $_SESSION['Username'] == $CreatedBy && $Reminder == '1' && $difference >= $ReminderDays && $Paid == '2')
		{
			$_SESSION['UserReminder'] = "1";
		}
	}
}

$smarty->assign("Title","$a[startpage]");
$smarty->assign("Welcome","$a[welcome]");
$smarty->assign("License","$a[license]");
$smarty->assign("OpenAccountMsg","$a[open_invoice]");
$smarty->assign("OpenSince","$a[open_since]");

// Get Update Status Information
//
$query1 = $db->Execute("SELECT LOGINUPDATE, TABLEUPDATE FROM updatetable WHERE UPDATEID=1");

if (!$query1)
	print($db->ErrorMsg());
else
	foreach($query1 as $f1)
	{
		$smarty->assign("LOGINUPDATE",$f1['LOGINUPDATE']);
		$smarty->assign("TABLEUPDATE",$f1['TABLEUPDATE']);
	}

$smarty->display('index.tpl');

// Delete the reminder session
//
unset ($_SESSION['UserReminder']);

?>
